### PR TITLE
distro/rhel86: use the new coreos-installer-dracut

### PR DESCRIPTION
**Pull request by runcom**
In 8.5 we're shipping patches on top of the coreos-installer rpm to enable it to run in the initramfs as a dracut module. That can easily become unmaintainable and introduces an hard dependency on coreos-installer itself. Since we're not modifying coreos-installer but just adding a dracut module, we've modified the coi rpm spec to introduce a new RPM `coreos-installer-initramfs` (`-dracut` was an old rpm and may conflict, will revisit later). This new RPM just ships the dracut module `51coi`  which is what we need in edge for the simplified installer. The new rpm is going to be in 8.6 so we'll need to find a sane way to avoid breaking current testing (I think).

This patch also drops coi from systemd enabled services as the installer system never actually boots but just runs in the initramfs.

cc @nullr0ute @gicmo @teg 

Signed-off-by: Antonio Murdaca <runcom@linux.com>

This pull request includes:

- [ ] adequate testing for the new functionality or fixed issue
- [ ] adequate documentation informing people about the change such as
  - [ ] create a file in [news/unreleased](https://github.com/osbuild/osbuild-composer/tree/main/docs/news/unreleased) directory if this change should be mentioned in the release news
  - [ ] submit a PR for the [guides](https://github.com/osbuild/guides) repository if this PR changed any behavior described there: https://www.osbuild.org/guides/

### CODE REVIEW

**Comment by runcom** [comment]
This is now rebased for 8.6

**Comment by runcom** [comment]
@teg @achilleas-k @gicmo the package now shows up in 8.6 - this can be merged

**Review by achilleas-k — DISMISSED** [review]

**Comment by achilleas-k** [comment]
Approved, but will need a rebase once #1949 gets merged

**Comment by runcom** [comment]
@achilleas-k does it needs a rebase or just porting these 8.6 changes over to 9?

**Comment by runcom** [comment]
Can somebody also clean up labels here

**Comment by achilleas-k** [comment]
> @achilleas-k does it needs a rebase or just porting these 8.6 changes over to 9?

The rebase referred to the failing RHEL 9.0 Beta tests, but that's taken care of here now #1982.
We'll have to port the changes to RHEL 9.0 as well, yes.  Maybe after it lands so as to not keep that open for even longer (assuming it goes green today).

**Comment by runcom** [comment]
@achilleas-k rebased now

File: test/data/repositories/rhel-86.json
Comment (runcom) [inline test/data/repositories/rhel-86.json:5]: seems like this snapshot still fails? https://gitlab.com/osbuild/ci/osbuild-composer/-/jobs/1791561513#L2698 @achilleas-k

Comment (runcom) [reply test/data/repositories/rhel-86.json:5]: seems like it's building on centos..... why? :thinking:

Comment (runcom) [reply test/data/repositories/rhel-86.json:5]: updated centos too

**Review by achilleas-k — APPROVED** [review]
Thanks so much for all your work on this!

**Pull request merged**
